### PR TITLE
feat(e2e): add e2e test for prometheus

### PR DESCRIPTION
**Pull request by Chever-John**
add e2e test for prometheus plugin.
#321 
https://github.com/apache/apisix/issues/6276

### Type of change:

- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Review by tao12345666333 — CHANGES_REQUESTED** [review]

File: test/e2e/e2e.coverprofile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+mode: atomic
```
Comment (tao12345666333) [inline test/e2e/e2e.coverprofile:1]: This file is not needed, please delete

**Review by tao12345666333 — CHANGES_REQUESTED** [review]

File: test/e2e/features/global_rule.go [hunk 2]
```diff
@@ -56,6 +58,40 @@ spec:
...
+     - httpbin.org
+     paths:
+       - /apisix/prometheus/metrics
+   backends:
+   - serviceName: %s
+     servicePort: %d
+     weight: 10
+   plugins:
+   - name: public-api
+     enable: true
```
Comment (tao12345666333) [inline test/e2e/features/global_rule.go:87]: The YAML here needs to be indented with two spaces

**Review by tao12345666333 — CHANGES_REQUESTED** [review]

File: test/e2e/features/global_rule.go [hunk 3]
```diff
@@ -56,6 +56,33 @@ spec:
...
+  name: httpbin-route
+spec:
+  http:
+  - name: rule1
+    match:
+      hosts:
+      - httpbin.org
+      paths:
+      - /apisix/prometheus/metrics
+  plugins:
```
Comment (tao12345666333) [inline test/e2e/features/global_rule.go:72]: plugins is a list and belong to `spec.http`

**Review by tao12345666333 — CHANGES_REQUESTED** [review]

File: test/e2e/features/global_rule.go [hunk 4]
```diff
@@ -64,8 +98,8 @@ spec:
 		_, ok := grs[0].Plugins["prometheus"]
 		assert.Equal(ginkgo.GinkgoT(), ok, true)
 
-		resp := s.NewAPISIXClient().GET("/apisix/prometheus/metrics").Expect()
-		resp.Status(200)
+		resp := s.NewAPISIXClient().GET("/apisix/prometheus/metrics").WithHeader("Host", "httpbin.org").Expect()
```
Comment (tao12345666333) [inline test/e2e/features/global_rule.go:101]: ditto. The name.

**Review by tao12345666333 — CHANGES_REQUESTED** [review]

File: test/e2e/plugins/server-info.go [hunk 5]
```diff
@@ -32,7 +32,7 @@ var _ = ginkgo.Describe("server-info plugin", func() {
 	}
 	s := scaffold.NewScaffold(opts)
 
-	ginkgo.It("enable server-info plugin", func() {
+	ginkgo.It("enable server-info plugin_", func() {
```
Comment (Chever-John) [inline test/e2e/plugins/server-info.go:35]: @tao12345666333 
I have run into an unexpected problem.
I just have modified the e2e test file for "global_rule". 
But it transferred the CI error reporting to the "server-info" plugin's e2e test file.

I have just tried as follows.
1. reinstall docker containers after run 
`sudo docker rm $(docker ps -a -q)` and 
`sudo rmi $(docker images -a)` and 
`sudo docker volume prune`.
2. and `git reset` to the version that we did not change the apisix-alpine(version is reseted to 2.12). There still be problem. 
3. The error message is `expect is True, actual is false`. (Sry, forgive me have delete all data and do not copy the former error message.

I am looking forward to hear from you!

Comment (tao12345666333) [reply test/e2e/plugins/server-info.go:35]: @Chever-John you can just ignore this or disable server-info plugin test case.

In APISIX v2.13 this plugin is disabled by default.

Comment (Chever-John) [reply test/e2e/plugins/server-info.go:35]: clear！

File: test/e2e/plugins/server-info.go [hunk 6]
```diff
@@ -32,13 +32,13 @@ var _ = ginkgo.Describe("server-info plugin", func() {
 	}
 	s := scaffold.NewScaffold(opts)
 
-	ginkgo.It("enable server-info plugin", func() {
+	ginkgo.It("disable server-info plugin", func() {
```
Comment (tao12345666333) [inline test/e2e/plugins/server-info.go:35]: There is no need to adjust the content of the test case, just Skip() the test case directly

**Review by tao12345666333 — APPROVED** [review]
LGTM. 

Let's move forward.

**Comment by tao12345666333** [comment]
@Chever-John  Could you please create another PR to fix the test case of the server-info plugin

File: test/e2e/features/global_rule.go [hunk 7]
```diff
@@ -56,6 +59,30 @@ spec:
 		// Wait until the ApisixClusterConfig create event was delivered.
 		time.Sleep(3 * time.Second)
 
+		ar := fmt.Sprintf(`
```
Comment (tokers) [inline test/e2e/features/global_rule.go:62]: I think it's worthy to write some documents about "How to access Apache APISIX Prometheus Metrics on Kubernetes".

Comment (tao12345666333) [reply test/e2e/features/global_rule.go:62]: There seems to be a similar article on APISIX's WeChat account

Comment (tokers) [reply test/e2e/features/global_rule.go:62]: Could we also file them?

Comment (tokers) [reply test/e2e/features/global_rule.go:62]: OK, let's do it in another PR.

Comment (tokers) [reply test/e2e/features/global_rule.go:62]: See https://github.com/apache/apisix-ingress-controller/issues/953.

**Review by tokers — APPROVED** [review]

**Review by lingsamuel — APPROVED** [review]

**Pull request merged**